ripe/atlas/tools/commands/report.py

```diff
@@ -161,4 +165,6 @@ def multi_level_render(self, aggregation_data, indent=""):
         elif isinstance(aggregation_data, list):
 
             for index, data in enumerate(aggregation_data):
-                self.payload = "{}{} {}".format(self.payload, indent, self.renderer.on_result(data))
+                res = self.renderer.on_result(data)
+                if res:
+                    self.payload = "{}{} {}".format(self.payload, indent, self.renderer.on_result(data))
```

Comment:
You're calling `self.renderer.on_result()` twice here.  Once to check if there's any output, and again to append it to the payload.  I'm not sure what the value of this bit is.

The restoration of the `on_start()` and `on_finish()` bits is awesome though.  Thanks for catching that.